File: API/Entidades/TipoDocumento.cs
using Barberiapp.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Barberiapp.Entidades
{
    public class TipoDocumento
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int CodigoTipoDocumento { get; set; }

        [Required(ErrorMessage = "El campo {0} es obligatorio")]
        [MaxLength(50, ErrorMessage = "El campo {0} no puede tener más de {1} caracteres")]
        public string Nombre { get; set; }

        // Referencias

        public ICollection<ApplicationUser > Usuarios { get; set; }
    }
}

File: APP/lib/models/tipo_documento.dart
class TipoDocumento {
  int codigoTipoDocumento;
  String nombre;

  TipoDocumento({
    required this.codigoTipoDocumento,
    required this.nombre,
  });
  factory TipoDocumento.fromJson(Map<String, dynamic> json) => TipoDocumento(
        codigoTipoDocumento: json["codigoTipoDocumento"],
        nombre: json["nombre"],
      );
  Map<String, dynamic> toJson() => {
        "codigoTipoDocumento": codigoTipoDocumento,
        "nombre": nombre,
      };
}